rust/src/lib.rs

```diff
@@ -2033,18 +2037,51 @@ pub extern "C" fn zcashlc_create_to_address(
 
         let prover = LocalTxProver::new(spend_params, output_params);
 
-        create_spend_to_address(
-            &mut db_data,
-            &network,
-            prover,
-            &usk,
-            &to,
-            value,
-            memo,
-            OvkPolicy::Sender,
-            min_confirmations,
-        )
-        .map_err(|e| format_err!("Error while sending funds: {}", e))
+        let req = TransactionRequest::new(vec![Payment {
+            recipient_address: to,
+            amount: value,
+            memo: memo,
+            label: None,
+            message: None,
+            other_params: vec![],
+        }])
+        .map_err(|e| format_err!("Error creating transaction request: {:?}", e))?;
+
+        if use_zip317_fees {
+            let input_selector = GreedyInputSelector::new(
+                zip317::SingleOutputChangeStrategy::new(Zip317FeeRule::standard()),
+                DustOutputPolicy::default(),
+            );
+
+            spend(
+                &mut db_data,
+                &network,
+                prover,
+                &input_selector,
+                &usk,
+                req,
+                OvkPolicy::Sender,
+                min_confirmations,
+            )
+            .map_err(|e| format_err!("Error while sending funds: {}", e))
+        } else {
+            let input_selector = GreedyInputSelector::new(
+                fixed::SingleOutputChangeStrategy::new(FixedFeeRule::standard()),
+                DustOutputPolicy::default(),
+            );
+
+            spend(
+                &mut db_data,
+                &network,
+                prover,
+                &input_selector,
+                &usk,
+                req,
+                OvkPolicy::Sender,
+                min_confirmations,
+            )
+            .map_err(|e| format_err!("Error while sending funds: {}", e))
+        }
```

Comment:
`InputSelector` is a trait, so why do we have to duplicate this?

Reply:
I see: [`zcash_client_backend::data_api::wallet::spend` is generic on `InputsT`](https://github.com/zcash/librustzcash/blob/bf6ad58a028ed8074e825f0f1c25e528b8041361/zcash_client_backend/src/data_api/wallet.rs#L280), and so the nearest you can get to avoiding duplication would be the code below (which is not even shorter, so I'm not suggesting it). Why not have `spend` take `Box<dyn InputSelector>`?

```rust
        fn spend_using<ChangeT: ChangeStrategy>(
            db_data: &mut DataConnStmtCache<Network>,
            network: &Network,
            prover: LocalTxProver,
            usk: &UnifiedSpendingKey,
            req: TransactionRequest,
            min_confirmations: u32,
            change_strategy: ChangeT,
        ) -> Result<i64, failure::Error>
        where
            <ChangeT as ChangeStrategy>::FeeRule: Clone,
            <<ChangeT as ChangeStrategy>::FeeRule as FeeRule>::Error: std::fmt::Display,
            <ChangeT as ChangeStrategy>::Error: std::fmt::Display,
        {
            spend(
                db_data,
                network,
                prover,
                &GreedyInputSelector::new(change_strategy, DustOutputPolicy::default()),
                usk,
                req,
                OvkPolicy::Sender,
                min_confirmations,
            )
            .map_err(|e| format_err!("Error while sending funds: {}", e))
        }

        if use_zip317_fees {
            spend_using(
                &mut db_data,
                &network,
                prover,
                &usk,
                req,
                min_confirmations,
                zip317::SingleOutputChangeStrategy::new(Zip317FeeRule::standard()),
            )
        } else {
            spend_using(
                &mut db_data,
                &network,
                prover,
                &usk,
                req,
                min_confirmations,
                fixed::SingleOutputChangeStrategy::new(FixedFeeRule::standard()),
            )
        }
```